src/Analysim.Web/ClientApp/src/app/register/register.component.ts

```diff
@@ -48,24 +60,43 @@ export class RegisterComponent implements OnInit {
       'emailAddress': this.emailAddress,
       'username': this.username,
       'password': this.password,
-      'confirmPassword': this.confirmPassword
+      'confirmPassword': this.confirmPassword,
+      'workPlace': this.workPlace,
+      'positionTitle': this.positionTitle,
+      'whereDidYouHearAboutAnalysim': this.whereDidYouHearAboutAnalysim,
+      'wouldYouLikeToHearUpdatesFromUs': this.wouldYouLikeToHearUpdatesFromUs,
+      'registrationCode': this.registrationCode,
     });
+
+    this.configService.checkRegistrationCodesLength().subscribe({
+      next: (data) => {
+        this.showHideRegistrationCode =  data === 0 ? false : true;
+      },
+      error: (error) => {
+        console.error(error);
+      }
+    })
+
   }
 
 
   onSubmit() {
     // Variable for FormGroupValue
     let userReg = this.insertForm.value
 
+    let registrationSurvey = userReg.workPlace+";"+userReg.positionTitle+";"+userReg.whereDidYouHearAboutAnalysim+";"+userReg.wouldYouLikeToHearUpdatesFromUs.toString()+";"+userReg.registrationCode;
```

Comment:
@KshitizSareen the only correction I will ask is to change this string to be proper JSON, such as:
```
' { "workPlace": ' + userReg.workPlace + ', ...'
```
This way we can parse it in the database as well.